TPCSpaceChargeBase/AliTPCSpaceCharge3DCalc.cxx

```diff
@@ -0,0 +1,4178 @@
+        matrixEz(i, j) = -1 * (arrayV(i, j + 1) - arrayV(i, j - 1)) / (2 * gridSizeZ); // z direction
+        matrixEPhi(i, j) = -1 * (signPlus * arrayVP(i, j) - signMinus * arrayVM(i, j)) /
+                           (2 * radius * gridSizePhi); // phi direction
+      }
+    }
+
+    // for boundary-r
+    for (Int_t j = 0; j < nZColumn; j++) {
+      matrixEr(0, j) = -1 * (-0.5 * arrayV(2, j) + 2.0 * arrayV(1, j) - 1.5 * arrayV(0, j)) /
+                       gridSizeR; // forward difference
+      matrixEr(nRRow - 1, j) =
+        -1 * (1.5 * arrayV(nRRow - 1, j) - 2.0 * arrayV(nRRow - 2, j) + 0.5 * arrayV(nRRow - 3, j)) /
+        gridSizeR; // backward difference
+    }
+
+    for (Int_t i = 0; i < nRRow; i += nRRow - 1) {
+      radius = innerRadius + i * gridSizeR;
+      for (Int_t j = 1; j < nZColumn - 1; j++) {
+        matrixEz(i, j) = -1 * (arrayV(i, j + 1) - arrayV(i, j - 1)) / (2 * gridSizeZ); // z direction
+        matrixEPhi(i, j) = -1 * (signPlus * arrayVP(i, j) - signMinus * arrayVM(i, j)) /
+                           (2 * radius * gridSizePhi); // phi direction
+      }
+    }
+
+    // for boundary-z
+    for (Int_t i = 0; i < nRRow; i++) {
+      matrixEz(i, 0) = -1 * (-0.5 * arrayV(i, 2) + 2.0 * arrayV(i, 1) - 1.5 * arrayV(i, 0)) / gridSizeZ;
+      matrixEz(i, nZColumn - 1) =
+        -1 *
+        (1.5 * arrayV(i, nZColumn - 1) - 2.0 * arrayV(i, nZColumn - 2) + 0.5 * arrayV(i, nZColumn - 3)) /
+        gridSizeZ;
+    }
+
+    for (Int_t i = 1; i < nRRow - 1; i++) {
+      radius = innerRadius + i * gridSizeR;
+      for (Int_t j = 0; j < nZColumn; j += nZColumn - 1) {
+        matrixEr(i, j) = -1 * (arrayV(i + 1, j) - arrayV(i - 1, j)) / (2 * gridSizeR); // r direction
+        matrixEPhi(i, j) = -1 * (signPlus * arrayVP(i, j) - signMinus * arrayVM(i, j)) /
+                           (2 * radius * gridSizePhi); // phi direction
+      }
+    }
+
+    // corner points for EPhi
+    for (Int_t i = 0; i < nRRow; i += nRRow - 1) {
+      radius = innerRadius + i * gridSizeR;
+      for (Int_t j = 0; j < nZColumn; j += nZColumn - 1) {
+        matrixEPhi(i, j) = -1 * (signPlus * arrayVP(i, j) - signMinus * arrayVM(i, j)) /
+                           (2 * radius * gridSizePhi); // phi direction
+      }
+    }
+  }
+}
+///
+/// Local distortion and correction, calculate local distortion/correction
+/// based on simplified langevin equation, see internal note ALICE-INT-2010-016.
+///
+/// <b> Local Distortion </b>
+///
+/// Local distortion is calculated based on formulation in ALICE-INT-2010-016, this function assume that
+/// electric field \f$\vec{E}(r_{i},z_{j},\phi_{m})\f$ is provided.
+///
+/// First, we calculate integration of the Electric field in z-direction for all direction.
+/// Assumption: \f$ z_{0} \f$ is location of CE (Central Electrode) and \f$ z_{nZColumn - 1} \f$ is location of End Plate.
+///
+/// This integration is in \f$z\f$ direction we can only 
```

Comment:
According to the convention, this should be
`-1 * dPhi * (radius0 + drDist)`
because the radius of the irregular point is radius0 + drDist.

Reply:
> According to the convention, this should be
> `-1 * dPhi * (radius0 + drDist)`
> because the radius of the irregular point is radius0 + drDist.

Hello Ernst, sorry I don't understand, so I need to change the line in 1498-1501 to your comment?